docs/README.md

```diff
@@ -0,0 +1,351 @@
+# Test and config files
+
+A typical `*-bench` project runs checks specified in `controls` files that are a YAML 
+representation of the CIS Benchmark checks. There is a 
+`controls` file per version and node type.
+
+`controls` for the various versions can be found in directories
+with same name as the CIS versions under `cfg/`, for example `cfg/cis-1.4.0`.
+`controls` are also organized by distribution under the `cfg` directory, for
+example `cfg/rh-0.7`.
+
+## Controls
+
+`controls` is a YAML document that contains checks that must be run against a 
+specific Kubernetes node type, master or node and version.
+
+`controls` is the fundamental input to a `*-bench` project. The following is a Kubernetes CIS Benchmark example 
+of a basic `controls`:
+
+```yml
+---
+controls:
+id: 1
+text: "Master Node Security Configuration"
+type: "master"
+groups:
+- id: 1.1
+  text: API Server
+  checks:
+    - id: 1.1.1
+      text: "Ensure that the --allow-privileged argument is set (Scored)"
+      audit: "ps -ef | grep kube-apiserver | grep -v grep"
+      tests:
+      bin_op: or
+      test_items:
+      - flag: "--allow-privileged"
+        set: true
+      - flag: "--some-other-flag"
+        set: false
+      remediation: "Edit the /etc/kubernetes/config file on the master node and
+        set the KUBE_ALLOW_PRIV parameter to '--allow-privileged=false'"
+      scored: true
+- id: 1.2
+  text: Scheduler
+  checks:
+    - id: 1.2.1
+      text: "Ensure that the --profiling argument is set to false (Scored)"
+      audit: "ps -ef | grep kube-scheduler | grep -v grep"
+      tests:
+        bin_op: or
+        test_items:
+          - flag: "--profiling"
+            set: true
+          - flag: "--some-other-flag"
+            set: false
+      remediation: "Edit the /etc/kubernetes/config file on the master node and
+        set the KUBE_ALLOW_PRIV parameter to '--allow-privileged=false'"
+      scored: true
+```
+
+`controls` is composed of a hierarchy of groups, sub-groups and checks. Each of
+the `controls` components have an id and a text description which are displayed 
+in the `*-bench` project output.
+
+`type` specifies what kubernetes node type a `controls` is for. Possible values
+for `type` are `master` and `node`.
+
+## Groups
+
+`groups` is list of subgroups which test the various Kubernetes components
+that run on the node type specified in the `controls`. 
```

Comment:
Kubernetes specific. I think we could make this generic by saying "`groups` represents a section in the benchmark. Depending on the benchmark a section (and thereby its group in the YAML definition` may represent a component, a group of components, or a set of functionality."